### PR TITLE
helloworld kathy

### LINKED ISSUE #536: Turn HelloWorld into kathy scripts
We need an easier way to generate messages on testnet. We should use the helloworld app to do so, waiting on #514

### PR DESCRIPTION
**Pull request by yorhodes**
- Add hello world commands to infra
Fixes #536

### CODE REVIEW

File: typescript/infra/helloworld.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (asaj) [inline typescript/infra/helloworld.json:1]: What is this file?

Comment (asaj) [reply typescript/infra/helloworld.json:1]: Are these contract addresses? Should we write them to `infra/config/environments/testnet2`?

File: typescript/infra/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@abacus-network/core": "^0.2.1-beta3",
     "@abacus-network/deploy": "^0.2.1-beta3",
+    "@abacus-network/helloworld": "0.2.1-alpha",
```
Comment (jmrossy) [inline typescript/infra/package.json:8]: See comment in your draft PR, would be good for this to match the SDK version it's using

File: typescript/infra/scripts/check-helloworld.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import {
+  HelloWorldApp,
+  HelloWorldChecker,
+  HelloWorldConfig,
+} from '@abacus-network/helloworld';
+import {
+  HelloWorldContracts,
+  helloWorldFactories,
+} from '@abacus-network/helloworld/dist/sdk/contracts';
```
Comment (jmrossy) [inline typescript/infra/scripts/check-helloworld.ts:9]: I was going to fix this in a follow-up PR but `sdk` isn't really a good name for what that folder is right? Feel free to ignore for this PR, just thinking out loud

File: typescript/infra/scripts/check-helloworld.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import addresses from '../config/environments/testnet2/helloworld/addresses.json';
+
+import { getCoreEnvironmentConfig } from './utils';
+
+async function main() {
+  const environment = 'testnet2';
+  const coreConfig = getCoreEnvironmentConfig(environment);
+  const multiProvider = await coreConfig.getMultiProvider();
+
+  type Chains = keyof typeof addresses;
```
Comment (jmrossy) [inline typescript/infra/scripts/check-helloworld.ts:26]: Nit: move to top

File: typescript/infra/scripts/deploy-helloworld.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }),
+  );
+
+  const deployer = new HelloWorldDeployer(multiProvider, configMap, core);
+  const contracts = await deployer.deploy();
+  const dir = path.join(getEnvironmentDirectory(environment), 'helloworld');
+  writeJSON(dir, 'addresses.json', serializeContracts(contracts));
+}
+
+main().then(console.log).catch(console.error);
```
Comment (jmrossy) [inline typescript/infra/scripts/deploy-helloworld.ts:42]: Nit: `then` block won't log anything

File: typescript/infra/scripts/kathy-helloworld.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    keyof typeof addresses,
+    HelloWorldContracts
+  >;
+  const app = new HelloWorldApp(contracts, multiProvider);
+  const sources = app.chains();
+  await Promise.all(
+    sources.map((source) => {
+      const destinations = sources.slice().filter((d) => d !== source);
+      return Promise.all(
+        destinations.map(async (destination) => {
```
Comment (jmrossy) [inline typescript/infra/scripts/kathy-helloworld.ts:26]: Double nested .map and promise.all, would be nice to break this apart a bit :)

File: typescript/infra/tsconfig.json [hunk 7]
```diff
@@ -1,7 +1,8 @@
 {
   "compilerOptions": {
     "outDir": "./dist/",
-    "rootDir": "./"
+    "rootDir": "./",
+    "resolveJsonModule": true
```
Comment (jmrossy) [inline typescript/infra/tsconfig.json:5]: We could probably add this to the root tsconfig

**Review by jmrossy — COMMENTED** [review]
Mostly looks good, just some small stuff

File: typescript/infra/scripts/check-helloworld.ts (hunk 3 above)
Comment (jmrossy) [reply typescript/infra/scripts/check-helloworld.ts:9]: Making a note for myself, will circle back on this

**Review by jmrossy — APPROVED** [review]

**Pull request merged**